be/src/exprs/array_functions.cpp

```diff
@@ -1688,4 +1688,49 @@ StatusOr<ColumnPtr> ArrayFunctions::repeat(FunctionContext* ctx, const Columns&
         return dest_column;
     }
 }
+
+StatusOr<ColumnPtr> ArrayFunctions::array_flatten(FunctionContext* ctx, const Columns& columns) {
+    DCHECK_EQ(1, columns.size());
+    RETURN_IF_COLUMNS_ONLY_NULL(columns);
+
+    size_t chunk_size = columns[0]->size();
+    Column* src_column = ColumnHelper::unpack_and_duplicate_const_column(chunk_size, columns[0]).get();
+
+    const NullableColumn* src_nullable_column = nullptr;
+    ArrayColumn* array_column = nullptr;
+    if (columns[0]->is_nullable()) {
+        src_nullable_column = down_cast<const NullableColumn*>(src_column);
+        array_column = down_cast<ArrayColumn*>(src_nullable_column->data_column().get());
+    } else {
+        array_column = down_cast<ArrayColumn*>(src_column);
+    }
+
+    auto [array_null, elements, offsets] = unpack_array_column(array_column->elements_column());
```

Comment:
it's better to add a check to ensure that the element column is also an array column to avoid unexpected problems.